libraries/Microsoft.Bot.Builder/TokenResolver.cs

```diff
@@ -65,10 +67,11 @@ private static async Task PollForTokenAsync(BotFrameworkAdapter adapter, ILogger
             }
 
             var stopwatch = Stopwatch.StartNew();
+            var oauthClient = turnContext.TurnState.Get<OAuthClient>();
```

Comment:
>oauthClient = turnContext.TurnState.Get<OAuthClient>(); [](start = 16, length = 55)

If this is null, we should probably revert to the built-in oauthClient.

Reply:
maybe that happens in the adapter?

---
In reply to: [356877905](https://github.com/microsoft/botbuilder-dotnet/pull/3035#discussion_r356877905) [](ancestors = 356877905)

Reply:
yeah we're not using the oauthClient directly. We're only using its Credentials property. If it's null the adapter internally (line 74) will use the credentials of the bot.